src/linear_algebra.jl

```diff
@@ -98,7 +98,22 @@ function solve_for(eq, var; simplify=false, check=true) # scalar case
     # the cases.
     a, b, islinear = linear_expansion(eq, var)
     check && @assert islinear
-    islinear || return nothing
+
+    if eq isa AbstractArray
+        for eqᵢ in eq
+            try
+                islinear &= Symbolics.isaffine(eqᵢ.lhs-eqᵢ.rhs, var)
+            catch e
+            end
```

Comment:
Instead of try/catch, the root issue should be handled.

Reply:
agreed. this is just a temporary fix. Want me to open an issue on the failed tests for isaffine?

Reply:
yes please